lib/banner_projects_helper_patch.rb

```diff
@@ -14,11 +14,10 @@ def self.included base # :nodoc:
 module ProjectsHelperMethodsBanner
   def project_settings_tabs_with_banner
     tabs = project_settings_tabs_without_banner
-    action = {:name => 'banner', 
+    tabs.push({:name => 'banner',
       :controller => 'banner',
-      :action => :show, 
-      :partial => 'banner/show', :label => :banner}
-    tabs << action if User.current.allowed_to?(action, @project)
-    tabs
+      :action => :manage_banner,
+      :partial => 'banner/show', :label => :banner})
+    tabs.select {|tab| User.current.allowed_to?(tab[:action], @project)}
```

Comment:
No, that's wrong.

A Plugin has no business messing with other tabs. The original code is correct. I can guarantee that this will actually break other plugins, not fix them. If this "fixes" anything, the bug is in one of the other plugins.
redmine_issue_checklists (redminecrm.com) used to do the wrong tabs.select() thing..and it broke things badly. I provided the fix, doing exactly what was here all along.